### PR TITLE
Solved all the TODOs and make sendMessage accept an json string

### PR DESCRIPTION
**Pull request by niansa**

### CODE REVIEW

File: discordpp/plugin-pretty.hh [hunk 1]
```diff
@@ -85,79 +186,65 @@ namespace discordpp{
...
-			}
+            }
+            std::clog << out << std::endl;
 			return out;
 		}
 	};
 
-	// https://discord.com/developers/docs/topics/gateway#update-status
+    // https://discord.com/developers/docs/topics/gateway#update-status
 	struct Status{
```
Comment (UndarkAido) [inline discordpp/plugin-pretty.hh:236]: Needs a validity check

File: discordpp/plugin-pretty.hh [hunk 2]
```diff
@@ -1,77 +1,178 @@
...
 		}
 	};
 
-	// https://discord.com/developers/docs/topics/gateway#activity-object
-	struct Activity{
-		bool active = true;
+    // https://discord.com/developers/docs/topics/gateway#activity-object
+    struct Activity: public apiStruct {
+        bool isValid() const{
+            return true;
```
Comment (UndarkAido) [inline discordpp/plugin-pretty.hh:167]: This can't be right?

Comment (niansa) [reply discordpp/plugin-pretty.hh:369]: Well it's perfectly fine as we have all the default values in here

Comment (niansa) [reply discordpp/plugin-pretty.hh:369]: It is, decause we have all the default values we need.

Comment (UndarkAido) [reply discordpp/plugin-pretty.hh:369]: How to you submit a status without an activity then?

Comment (niansa) [reply discordpp/plugin-pretty.hh:369]: I have seen no problem in that

Comment (UndarkAido) [reply discordpp/plugin-pretty.hh:369]: How do you do it then?

Comment (UndarkAido) [reply discordpp/plugin-pretty.hh:369]: `name` is required and doesn't have a default value

**Comment by niansa** [comment]
I more or less stopped my work on here, because the code quality I produced here is simply bad and some stuff is solved in a dumb way (see the bitset, I should have simply used std::bitset instead).
Additionally, I lost interest in Discord bots.

**Comment by UndarkAido** [comment]
All good. Best of luck in future endeavors!

**Pull request closed without merging**